controllers/common/helper.go

```diff
@@ -694,3 +696,32 @@ func GetRecommendedCRVersion(
 	}
 	return nil, nil //nolint:nilnil
 }
+
+// DeploymentIsReady returns true if the deployment is ready.
+func DeploymentIsReady(ctx context.Context, c client.Client, name types.NamespacedName) (bool, error) {
+	deployment := &appsv1.Deployment{}
+	err := c.Get(ctx, name, deployment)
+	if err != nil {
+		return false, err
+	}
+
+	return deployment.Status.ReadyReplicas == *deployment.Spec.Replicas, nil
+}
+
+// RestartDeployment restarts the deployment.
+func RestartDeployment(
+	ctx context.Context,
+	c client.Client,
+	name types.NamespacedName,
+) error {
+	deployment := &appsv1.Deployment{}
+	err := c.Get(ctx, name, deployment)
+	if err != nil {
+		return err
+	}
+	if deployment.Spec.Template.ObjectMeta.Annotations == nil {
+		deployment.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+	}
+	deployment.Spec.Template.ObjectMeta.Annotations["kubectl.kubernetes.io/restartedAt"] = metav1.Now().Format(time.RFC3339)
+	return c.Update(ctx, deployment)
+}
```

Comment:
This annotation needs to come in through the CSV, not the Deployment, otherwise the CSV can override the Deployment annotations before the deployment-controller has had a chance to observe the updated annotation.

This is what we do in the Everest server, I already faced this problem once 😅 - https://github.com/percona/everest/blob/02f5f93e33b9ab9803d01bbb7715869fc4fbfdef/pkg/kubernetes/kubernetes.go#L838

Reply:
But then our dev setup will be broken since we don't use OLM there 🤷🏻